lib/resty/core/base64.lua

```diff
@@ -47,7 +47,7 @@ ngx.encode_base64 = function (s, no_padding)
 
     if no_padding then
         if no_padding ~= true then
-            return error("boolean argument only")
+            error("boolean argument only")
```

Comment:
We will need to make use of the second argument to `error` here (and in many other places where it makes sense) so the error is raised at the correct stack level. The previous approach with `return` would tail call and incidentally report the correct stack trace previously.

Here (and in other places where the error needs to appear as originating from the user's code stack level):
```
error("boolean argument only", 2)
```